packages/enzyme-test-suite/test/selector-spec.jsx

```diff
@@ -354,6 +362,74 @@ describe('selectors', () => {
+
+      it('should parse numeric literals', () => {
+        expectAttributeMatch(<div data-foo={2.3} />, '[data-foo=2.3]', true);
+        expectAttributeMatch(<div data-foo={2} />, '[data-foo=2]', true);
+        expectAttributeMatch(<div data-foo={2} />, '[data-foo="2abc"]', false);
+        expectAttributeMatch(<div data-foo={2} />, '[data-foo="abc2"]', false);
+        expectAttributeMatch(<div data-foo={-2} />, '[data-foo=-2]', true);
+        // @TODO this is failing due to a parser issue
+        // expectAttributeMatch(<div data-foo={2e8} />, '[data-foo=2e8]', true);
+        expectAttributeMatch(<div data-foo={Infinity} />, '[data-foo=Infinity]', true);
+        expectAttributeMatch(<div data-foo={Infinity} />, '[data-foo=-Infinity]', false);
+        expectAttributeMatch(<div data-foo={-Infinity} />, '[data-foo=-Infinity]', true);
+        expectAttributeMatch(<div data-foo={-Infinity} />, '[data-foo=Infinity]', false);
+      });
+
+      it('should parse zeroes properly', () => {
+        expectAttributeMatch(<div data-foo={0} />, '[data-foo=0]', true);
+        expectAttributeMatch(<div data-foo={0} />, '[data-foo=+0]', true);
+        expectAttributeMatch(<div data-foo={-0} />, '[data-foo=-0]', true);
+        expectAttributeMatch(<div data-foo={-0} />, '[data-foo=0]', false);
+        expectAttributeMatch(<div data-foo={0} />, '[data-foo=-0]', false);
+        expectAttributeMatch(<div data-foo={1} />, '[data-foo=0]', false);
+        expectAttributeMatch(<div data-foo={2} />, '[data-foo=-0]', false);
+      });
+
+      it('should work with empty strings', () => {
+        expectAttributeMatch(<div className="" />, '[className=""]', true);
+        expectAttributeMatch(<div className={''} />, '[className=""]', true);
+        expectAttributeMatch(<div className={'bar'} />, '[className=""]', false);
+      });
+
+      it('should work with NaN', () => {
+        expectAttributeMatch(<div data-foo={NaN} />, '[data-foo=NaN]', true);
+        expectAttributeMatch(<div data-foo={0} />, '[data-foo=NaN]', false);
+      });
+
+      it('should work with null', () => {
+        expectAttributeMatch(<div data-foo={null} />, '[data-foo=null]', true);
+        expectAttributeMatch(<div data-foo={0} />, '[data-foo=null]', false);
+      });
+
+      it('should work with false', () => {
+        expectAttributeMatch(<div data-foo={false} />, '[data-foo=false]', true);
+        expectAttributeMatch(<div data-foo={0} />, '[data-foo=false]', false);
+      });
+      it('should work with ±Infinity', () => {
+        expectAttributeMatch(<div data-foo={Infinity} />, '[data-foo=Infinity]', true);
+        expectAttributeMatch(<div data-foo={Infinity} />, '[data-foo=+Infinity]', true);
+        expectAttributeMatch(<div data-foo={Infinity} />, '[data-foo=-Infinity]', false);
+        expectAttributeMatch(<div data-foo={Infinity} />, '[data-foo=NaN]', false);
+        expectAttributeMatch(<div data-foo={0} />, '[data-foo=Infinity]', false);
```

Comment:
It looks like this test is duplicated at the bottom of the `it` block - should one of them be testing `-Infinity`, as per the old tests?

Reply:
I think so, thanks!